pkg/types/main_test.go

```diff
@@ -37,3 +43,172 @@ func TestMainInitNodeID(t *testing.T) {
 		})
 	}
 }
+
+func TestNodeCfgInitEmptyID(t *testing.T) {
+	originalHostname := ""
+	defer func() {
+		if originalHostname != "" {
+			os.Setenv("HOSTNAME", originalHostname)
+		}
+	}()
```

Comment:
originalHostname var and defer func doesn't seem to needed. I might be missing something but I removed it and the test functions with no issues.